app/scripts/controllers/deployments.js

```diff
@@ -28,119 +39,151 @@ angular.module('openshiftConsole')
+          LabelFilter.setLabelSuggestions($scope.labelSuggestions);
+          $scope.k8sDeployments = LabelFilter.getLabelSelector().select($scope.unfilteredK8SDeployments);
+          Logger.log("k8sDeployments (subscribe)", $scope.unfilteredK8SDeployments);
+        }));
+
+        function updateFilterWarning() {
+          var isFiltering = !LabelFilter.getLabelSelector().isEmpty();
+          var isFilteringAllDCs = $.isEmptyObject($scope.deploymentConfigs) && !$.isEmptyObject($scope.unfilteredDeploymentConfigs);
+          var thereAreDCs = !$.isEmptyObject($scope.unfilteredDeploymentConfigs);
+          var isFilteringAllRCs = $.isEmptyObject($scope.deploymentsByDeploymentConfig['']) && !$.isEmptyObject($scope.unfilteredReplicationControllers);
+          var thereAreRCs = !$.isEmptyObject($scope.unfilteredReplicationControllers);
+          var isFilteringAllK8SDeployments = $.isEmptyObject($scope.k8sDeployments) && !$.isEmptyObject($scope.unfilteredK8SDeployments);
+          var thereAreK8SDeployments = !$.isEmptyObject($scope.unfilteredK8SDeployments);
+          var isFilteringAllReplicaSets = $.isEmptyObject($scope.replicaSets) && !$.isEmptyObject($scope.unfilteredReplicaSets);
+          var thereAreReplicaSets = !$.isEmptyObject($scope.unfilteredReplicaSets);
+
+          // TODO: Update warnings for k8s deployments and replica sets.
+          if (isFiltering &&
+              (isFilteringAllDCs || !thereAreDCs) &&
+              (isFilteringAllRCs || !thereAreRCs) &&
+              (isFilteringAllK8SDeployments || !thereAreK8SDeployments) &&
+              (isFilteringAllReplicaSets || !thereAreReplicaSets) &&
+              (thereAreDCs || thereAreRCs || thereAreK8SDeployments || thereAreReplicaSets)) {
+            $scope.alerts["deployments"] = {
+              type: "warning",
+              details: "The active filters are hiding all deployments."
+            };
+          }
+          else {
+            delete $scope.alerts["deployments"];
+          }
+        }
+
+        $scope.showEmptyMessage = function() {
+          if ($filter('hashSize')($scope.deploymentsByDeploymentConfig) === 0) {
+            return true;
+          }
+
+          if ($filter('hashSize')($scope.deploymentsByDeploymentConfig) === 1 && $scope.deploymentsByDeploymentConfig['']) {
+            return true;
+          }
+
+          return false;
+        };
+
+        LabelFilter.onActiveFiltersChanged(function(labelSelector) {
+          // trigger a digest loop
+          $scope.$apply(function() {
+            $scope.deploymentConfigs = labelSelector.select($scope.unfilteredDeploymentConfigs);
+            $scope.deploymentsByDeploymentConfig = DeploymentsService.associateDeploymentsToDeploymentConfig($scope.deployments, $scope.deploymentConfigs, true);
+            if ($scope.deploymentsByDeploymentConfig['']) {
+              $scope.unfilteredReplicationControllers = $scope.deploymentsByDeploymentConfig[''];
```

Comment:
Looks like we have two deployments service methods that do the association differently.